core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -58,11 +68,211 @@ case class PSBT(
       vec.head.version
     }
   }
+
+  /**
+    * @param tx Transaction to add to PSBT
+    * @param i index of inputMap to add tx to
+    * @return PSBT with added tx
+    */
+  def addTransaction(tx: Transaction, i: Int): PSBT = {
+    require(
+      i < inputMaps.size,
+      s"i must be less than the number of input maps present in the psbt, $i >= ${inputMaps.size}")
+    if (inputMaps(i).isFinalized) {
+      this
```

Comment:
Shouldn't you ensure that the `tx` doesn't conflict with `this.inputMaps` and throw if it does?